types/milesight/EM410-RDL/uplink.js

```diff
@@ -0,0 +1,539 @@
+      if (signalQualityResult === 0) {
+        decoded.signalQuality = readInt16LE(bytes.slice(offset, offset + 2));
+      }
+      offset += 3;
+      break;
+    }
+    case 0x15: {
+      // distance_threshold_sensitive
+      const distanceThresholdSensitiveResult = readUInt8(bytes[offset + 2]);
+      if (distanceThresholdSensitiveResult === 0) {
+        decoded.distanceThresholdSensitive =
+          readInt16LE(bytes.slice(offset, offset + 2)) / 10;
+      }
+      offset += 3;
+      break;
+    }
+    case 0x16: {
+      // peak_sorting
+      const peakSortingResult = readUInt8(bytes[offset + 1]);
+      if (peakSortingResult === 0) {
+        decoded.peakSorting = readUInt8(bytes[offset]);
+      }
+      offset += 2;
+      break;
+    }
+    case 0x0d: {
+      // retransmit_config
+      const retransmitConfigResult = readUInt8(bytes[offset + 3]);
+      if (retransmitConfigResult === 0) {
+        decoded.retransmitConfig = {};
+        decoded.retransmitConfig.enable = readUInt8(bytes[offset]);
+        decoded.retransmitConfig.retransmitInterval = readUInt16LE(
+          bytes.slice(offset + 1, offset + 3),
+        );
+      }
+      offset += 4;
+      break;
+    }
+    case 0x39: {
+      // collection_interval
+      const collectionIntervalResult = readUInt8(bytes[offset + 2]);
+      if (collectionIntervalResult === 0) {
+        decoded.collectionInterval = readUInt16LE(
+          bytes.slice(offset, offset + 2),
+        );
+      }
+      offset += 3;
+      break;
+    }
+    default:
+      throw new Error("UNKNOWN_DOWNLINK_RESPONSE");
+  }
+
+  return { data: decoded, offset };
+}
+
+if (!Object.assign) {
+  Object.defineProperty(Object, "assign", {
+    enumerable: false,
+    configurable: true,
+    writable: true,
+    value(target) {
+      if (target == null) {
+        // TypeError if undefined or null
+        throw new TypeError("CANNOT_CONVERT_FIRST_ARGUMENT_TO_OBJECT");
+      }
+
+      const to = Object(target);
+      for (let i = 1; i < arguments.length; i++) {
+        let nextSource = arguments[i];
+        if (nextSource == null) {
+          // Skip over if undefined or null
+          continue;
+        }
+        nextSource = Object(nextSource);
+
+        const keysArray = Object.keys(Object(nextSource));
+        for (
+          let nextIndex = 0, len = keysArray.length;
+          nextIndex < len;
+          nextIndex++
+        ) {
+          const nextKey = keysArray[nextIndex];
+          const desc = Object.getOwnPropertyDescriptor(nextSource, nextKey);
+          if (desc !== undefined && desc.enumerable) {
+            to[nextKey] = nextSource[nextKey];
+          }
+        }
+      }
+      return to;
+    },
+  });
+}
+
+function isEmpty(obj) {
+  if (obj === undefined) {
+    return true;
+  }
+  return Object.keys(obj).length === 0;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const bytes = Hex.hexToBytes(payload);
+  const decoded = {};
```

Comment:
define the variable in the scope that it is used to prevent side-effects from previous iterations

Reply:
Pretty sure its a variable message size depending on your sensor settings. So i do think its placed correctly there.